File: Java/javaperformancetester/src/main/java/edu/imepac/javaperformancetester/services/VagaoService.java
package edu.imepac.javaperformancetester.services;

import edu.imepac.javaperformancetester.models.Vagao;
import edu.imepac.javaperformancetester.repositories.VagaoRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;

import java.util.List;

@Service
public class VagaoService {

    @Autowired
    private VagaoRepository vagaoRepository;

    @GetMapping
    public List<Vagao> findAll(){
        return vagaoRepository.findAll();
    }

    @PostMapping
    public Vagao create(@RequestBody Vagao vagao){
        return vagaoRepository.save(vagao);
    }

}

File: Kotlin/kotlinperformancetester/src/main/kotlin/edu/imepac/kotlinperformancetester/services/VagaoService.kt
package edu.imepac.kotlinperformancetester.services

import edu.imepac.kotlinperformancetester.models.Vagao
import edu.imepac.kotlinperformancetester.repositories.VagaoRepository
import jakarta.transaction.Transactional
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class VagaoService {
    @Autowired
    private lateinit var vagaoRepository: VagaoRepository

    fun findAll(): List<Vagao> {
        return vagaoRepository.findAll()
    }

    @Transactional
    fun salvarTodos(vagoes: List<Vagao>) : List<Vagao> {
        return vagaoRepository.saveAll(vagoes)
    }
    fun create(vagao: Vagao): Vagao {
        return vagaoRepository.save(vagao)
    }
}